### ISSUE TITLE
use scroll wheel to change vars inflight

### ISSUE THREAD

**Issue by strgaltdel** [FrSkyRC/ETHOS-Feedback-Community#4879]
**What options do we currently have for adjusting (trim) parameters in-flight?**

(e.g., differential or elevator mixing with butterfly, motor, flaps, etc.)
Best practice is to assign the corresponding parameters as vars and adjust them using actions through appropriate inputs.
Particularly noteworthy is the "repurpose trims" function, which (along with the wide range of possible vars) brought a significant improvement.
Trim inputs are currently the only practical option for the aforementioned goal.

The downside is that you need several trim buttons to adjust multiple vars or additional switching levels to use one or three trimmers to adjust multiple vars via trimmer, which can quickly become confusing.
If you want to adjust a wide value-range, it takes some getting used to, as you rely on the progressive scaling of step size.

Potentiometers (pots) would also make good inputs and provide "direct feedback" depending on their position. However, a pot cannot easily be switched between vars or FMs because it must be ensured that the pot is neutral when switching. Otherwise, unexpected changes to the var value could occur if the pot is accidentally turned to full range.

**The ideal input medium would be a rotary encoder ("scroll wheel"),** as it can always pick up from the current value when switching between flight modes or vars.
I had hoped this year that an X24 with this "feature" might be released, but unfortunately, that’s not happening, and such a transmitter is not on the horizon.

**Idea:**
Why not use the existing encoder?
Both, on my X18 and X20, are easy to operate "blind."
It just needs to ensure that its "normal function" is disabled during trimming.

**Proposal:**
A special function called "repurpose scroll wheel" will be introduced 
When activated, the "normal function" of the wheel is disabled.

- (1) Movements of the scroll wheel remain accessible via Lua/event handler.
- (2) Within vars, an action "repurpose wheel" can be used, which is activated by a condition. This condition is valid whenever both the condition and the SF "repurpose scroll wheel" are active.

Further explanation of (2):
Why are both the SF and a condition necessary?
This enables several different variables to be selected and adjusted during flight:

- a)  A var is used as a counter via actions, e.g., T5 can count the var up and down from 0 to 10.
- b)  For each count value, exactly one logical switch (LSW) becomes active.
- c)  When SF "Repurpose wheel" is active and a corresponding LSW is active, the respective VAR can be trimmed using the wheel in the currently active flight mode, FMs can be changed during this operation.
- d)  When SF "Repurpose wheel" is active and the corresponding LSW (depending on the count value) becomes active, an SF can trigger an announcement for the respective var (e.g., "Differential," "Elevator to flap," "Expo aileron," etc.). This LSW could get some delay, so you can "jump" over some vars without hearing  the call.

**In short:**
Only one additional trim input (to control the counter) is necessary to trim multiple vars in-flight by getting a good haptic feedback from what you are doing...

btw
by opening this issue #4300 comes into mind again.
even in 1.6 vars are propagated with delay...

regards

Udo

**Comment by 3wintro** [comment]
I think it's a very good idea. So you could fly everything on the first flight and trim everything. There used to be a similar LUA where you could do this, unfortunately it no longer works from 1.5.0. It was very helpful.

**Comment by glipski50** [comment]
Great idea. Glider pilots with many flight phases would appriciate that for sure.

**Comment by Wolepo** [comment]
You could make good use of that. Especially in the VARs or wherever larger areas need to be painted over quickly and precisely, the Scoll wheel would be well suited for this.

**Comment by strgaltdel** [comment]
... "ping"

**Comment by flyingeek** [comment]
Nice idea !

**Comment by Toffs** [comment]
Having programmed the X20’s 6 buttons to give me 9 different ‘selectors’ for a spare trim selector to edit 9 different VARS (with vocal readouts to help remember which does what)… I’m in favour!
